backend/models.py

```diff
@@ -57,8 +57,8 @@ class Endpoint(models.Model):
     # Additional JSON configuration object
     agent_cfg = models.JSONField(blank=True, null=True)
     # What other endpoints does this endpoint have direct access to?
-    upstream_connections = models.ManyToManyField("self")
-    downstream_connections = models.ManyToManyField("self")
+    # this may be wrong according to https://stackoverflow.com/questions/39821723/django-rest-framework-many-to-many-field-related-to-itself

```

Comment:
consider leading this comment with a `FIXME:` or a `TODO:` or smth, makes it easier to find if it turns out we have some inexplicable problem later on